Add ZImageDenoiserLoopModule

ZiniuLin opened the pull request:
## Main Change:
1. Add ZImageDenoiserLoopModule to module genai pipeline
2. Add ZImageDenoiserLoopModule test with python ov text encoder and vae decoder.

## Test Result:
<img width="405" height="478" alt="module test result" src="https://github.com/user-attachments/assets/6eba1017-6cbc-4984-a582-76c7d87ef380" />

<img width="512" height="512" alt="denoiser loop python test result" src="https://github.com/user-attachments/assets/e09935a4-b85a-4664-a541-b6a96a3761b4" />

ZiniuLin committed:
Add Transformer module to module pipeline

Add Transformer module to module pipeline.

Signed-off-by: Ziniu Lin <ziniu.lin@intel.com>

ZiniuLin committed:
Add zimage transformer python test with optimum-intel

Add zimage transformer python test with optimum-intel.
This test uses text encoder and vae decoder from optimum-intel,
and the transformer is from module genai pipeline.

Signed-off-by: Ziniu Lin <ziniu.lin@intel.com>

ZiniuLin committed:
Fix params auto padding issue when multiple inputs exist

Fix params auto padding issue when multiple inputs exist.

Signed-off-by: Ziniu Lin <ziniu.lin@intel.com>

ZiniuLin committed:
Rename TransformerModule to ZImageDenoiserLoopModule

Rename TransformerModule to ZImageDenoiserLoopModule.

Signed-off-by: Ziniu Lin <ziniu.lin@intel.com>

xipingyan commented on samples/cpp/module_genai/ut_modules/ZImageDenoiserLoopModule.cpp:
```diff
@@ -0,0 +1,61 @@
+        source: "pipeline_params.prompt_embed"
+      - name: "num_inference_steps"
+        type: "Int"
+        source: "pipeline_params.num_inference_steps"
```
Now we support not to provide source when only providing one Module in `pipeline_modules`.

No update is need, just a reminder.

xipingyan commented on samples/cpp/module_genai/ut_modules/ZImageDenoiserLoopModule.cpp:
```diff
@@ -0,0 +1,61 @@
+
+        auto prompt_embed = ut_randn_tensor(ov::Shape{101, 2560}, 42);
+        inputs["prompt_embed"] = prompt_embed;
+        inputs["num_inference_steps"] = 5;
```
```suggestion
        inputs["num_inference_steps"] = 2;
```
Just reduce unit time and cover multiple inferences.

xipingyan commented on src/cpp/src/image_generation/numpy_utils.hpp:
```diff
@@ -51,6 +51,10 @@ ov::Tensor concat(ov::Tensor tensor_1, ov::Tensor tensor_2, int axis);
+ov::Tensor squeeze(const ov::Tensor& tensor, size_t dim);
+ov::Tensor unsqueeze(const ov::Tensor& tensor, size_t dim);
+std::vector<ov::Tensor> split(const ov::Tensor& tensor);
+ov::Tensor stack(const std::vector<ov::Tensor>& tensors);
```
Great to provide these utils function, 
but I still suggest to move them to modules_genai's utils, because I will reduce rebase conflict when we rebase codes in the future. 
What do you think？

xipingyan reviewed (approved):
Please update and then merge.

ZiniuLin committed:
Move tensor operations to other file

Move tensor operations to other file and reduce
the inference steps of z image denoiser test.

Signed-off-by: Ziniu Lin <ziniu.lin@intel.com>

xipingyan commented on samples/cpp/module_genai/ut_modules/ZImageDenoiserLoopModule.cpp:
```diff
@@ -0,0 +1,57 @@
+    inputs:
+      - name: "prompt_embed"
+        type: "OVTensor"
+      - name: "num_inference_steps"
```
```suggestion
      - name: "num_inference_steps"    # [optional, default 9
```

ZiniuLin replied:
optional is added.

xipingyan commented on samples/cpp/module_genai/ut_modules/ZImageDenoiserLoopModule.cpp:
```diff
@@ -0,0 +1,57 @@
+      - name: "width"
+        type: "Int"
+      - name: "height"
+        type: "Int"
```
```suggestion
      - name: "width"    # [optional], default 1024
        type: "Int"
      - name: "height"   # [optional], default 1024
        type: "Int"
```

ZiniuLin replied:
optional is added.
